inbox/headless/api-reference.mdx

```diff
@@ -0,0 +1,331 @@
+#### Params
+<ParamField path="tags" type="string[]"/>
+#### Usage
+```typescript
+await novu.notifications.archiveAll({
+  tags: ["tag1", "tag2"],
+});
+```
+
+### archiveAllRead
+`archiveAllRead` method marks all read notifications as archived. Notifications can be filtered by tags.
+#### Params
+<ParamField path="tags" type="string[]"/>
+#### Usage
+```typescript
+await novu.notifications.archiveAllRead({
+  tags: ["tag1", "tag2"],
+});
+```
+
+### completePrimary
+`completePrimary` method marks primary action of a notification as completed. It changes the `isCompleted` field of `primaryAction` to **done** from **pending**.
+#### Params
+<ParamField path="notificationId" type="string"/>
+#### Usage
+```typescript
+await novu.notifications.completePrimary({
+  notificationId: "inbox_notification_id",
+});
+```
+
+### completeSecondary
+`completeSecondary` method marks secondary action of a notification as completed. It changes the `isCompleted` field of `primaryAction` to **done** from **pending**.
+#### Params
+<ParamField path="notificationId" type="string"/>
+#### Usage
+```typescript
+await novu.notifications.completeSecondary({
+  notificationId: "inbox_notification_id",
+});
+```
+
+### revertPrimary
+`revertPrimary` method marks primary action of a notification as pending. It changes the `isCompleted` field of `primaryAction` to **pending** from **done**.
+#### Params
+<ParamField path="notificationId" type="string"/>
+#### Usage
+```typescript
+await novu.notifications.revertPrimary({
+  notificationId: "inbox_notification_id",
+});
+```
+
+### revertSecondary
+`revertSecondary` method marks secondary action of a notification as pending. It changes the `isCompleted` field of `secondaryAction` to **pending** from **done**.
+#### Params
+<ParamField path="notificationId" type="string"/>
+#### Usage
+```typescript
+await novu.notifications.revertSecondary({
+  notificationId: "inbox_notification_id",
+});
+```
+
+## Preferences
+### Methods
+- [list](#list)
+- [update](#update)
+
+### list
+`list` method fetch the global and workflows channel preferences of the subscriber.
+#### Usage
+```typescript
+await novu.preferences.list()
+```
+
+### update
+`update` method updates the channel preferences of the subscriber. It updates the channel preferences of the subscriber for the specific workflow if `workflowId` is provided. If `workflowId` is not provided, it updates the global channel preferences of the subscriber. Multiple channels can be updated at once.
+#### Params
+<ParamField path="workflowId" type="string"/> 
+<ParamField path="channelPreferences" type="ChannelPreference"/> 
+
+```typescript ChannelPreference
+type ChannelPreference = {
+  email?: boolean;
+  sms?: boolean;
+  in_app?: boolean;
+  chat?: boolean;
+  push?: boolean;
+};
+```
+#### Usage
+```typescript
+await novu.preferences.update({
+  workflowId: "workflow_id",
+  channelPreferences: { email: false, sms: false },
+});
+```
+
+## Events
```

Comment:
We should explain here how the events work and what kind of data and when they return.